third_party/terraform/resources/resource_kms_crypto_key.go

```diff
@@ -242,12 +254,20 @@ func resourceKmsCryptoKeyDelete(d *schema.ResourceData, meta interface{}) error
 [WARNING] KMS CryptoKey resources cannot be deleted from GCP. The CryptoKey %s will be removed from Terraform state,
 and all its CryptoKeyVersions will be destroyed, but it will still be present on the server.`, cryptoKeyId.cryptoKeyId())
 
+	// Delete all versions of the key
 	err = clearCryptoKeyVersions(cryptoKeyId, config)
-
 	if err != nil {
 		return err
 	}
 
+	// Make sure automatic key rotation is disabled.
+	err = disableCryptoKeyRotation(cryptoKeyId, config)
+	if err != nil {
```

Comment:
nit: I really like the if-with-initialization-statement syntax (`if err = disableStuff(); err != nil {`), but totally up to you (if you replace this with that, I'd ask to also do it on line 258)